auto_causality/scoring.py

```diff
@@ -44,6 +46,22 @@ def const_marginal_effect(self, X):
 #         cate_estimate > 0,
 #     )
 #     return erupt_score
+def energy_distance_scores(
+    estimate: CausalEstimate,
+    df: pd.DataFrame,
+):
+    est = estimate.estimator
+    treatment_name = est._treatment_name[0]
+    dy = estimate.estimator.effect(df)
+    df["yhat"] = dy - df[est._outcome_name]
+
+    t1 = df[df[treatment_name] == 1]
+    t0 = df[df[treatment_name] == 0]
```

Comment:
This is wrong, you split not by treated or not treated, but by whether feature was available (so value of instrumental variable)